Building collectd --with-libgcrypt fails

*   Version of collectd: 5.7.1
*   Operating system / distribution: Debian

## Expected behavior

I expect `./configure --with-libgrcrypt && make` to succeed assuming the required deps are properly installed.

## Actual behavior

Upon running make: 

```
...
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_close'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_close'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_setkey'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_read'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_encrypt'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_check_version'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_randomize'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_write'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_open'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_open'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_setkey'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_setiv'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_control'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_cipher_ctl'
libcollectdclient/.libs/libcollectdclient.so: undefined reference to `gcry_md_hash_buffer'
```

## Steps to reproduce

I am attempting to compile collectd 5.7.1 with libgcrypt enabled for secure networking. I initially installed `libgcrypt11 libgcrypt11-dev` via `apt` though could not build because a newer of version of `libgcrypt` is required. I then moved onto install `libgcrypt` from source by running

```
wget https://www.gnupg.org/ftp/gcrypt/libgpg-error/libgpg-error-1.27.tar.bz2
tar -xjf libgpg-error-1.27.tar.bz2
cd libgpg-error-1.27/
./configure
make
make install
cd -
rm libgpg-error-1.27.tar.bz2
rm -r libgpg-error-1.27/

wget https://www.gnupg.org/ftp/gcrypt/libgcrypt/libgcrypt-1.8.0.tar.bz2
tar -xjf libgcrypt-1.8.0.tar.bz2
cd libgcrypt-1.8.0/
./configure
make 
make install
cd -
rm libgcrypt-1.8.0.tar.bz2
rm -r libgcrypt-1.8.0/
```

This installed `libgcrypt.la  libgcrypt.so  libgcrypt.so.20  libgcrypt.so.20.2.0  libgpg-error.la  libgpg-error.so  libgpg-error.so.0  libgpg-error.so.0.22.0` to my `/usr/local/lib`.

I then attempted to build collectd by running 

```
wget https://storage.googleapis.com/collectd-tarballs/collectd-5.7.1.tar.bz2
tar -xjf collectd-5.7.1.tar.bz2
cd collectd-5.7.1
./configure --with-libgcrypt
make
```

This results in the errors posted above.

While running `./configure` this prints to terminal `checking for the default library search path... /lib /usr/lib /usr/local/lib /lib/x86_64-linux-gnu /usr/lib/x86_64-linux-gnu`. It appears that `/usr/local/lib` is on the search path. 

I have tried forcing / ensuring `/usr/local/lib` is linked against via `LDFLAGS=-L/usr/local/lib/ CPPFLAGS=-I/usr/local/lib/ CFLAGS=-I/usr/local/lib/ ./configure --with-libgcrypt` though this has not worked. 

I also noticed the `Makefile.in` contains the following. Maybe these should be set somehow? 

```
GCRYPT_CPPFLAGS = @GCRYPT_CPPFLAGS@
GCRYPT_LDFLAGS = @GCRYPT_LDFLAGS@
GCRYPT_LIBS = @GCRYPT_LIBS@
```

I think it's a bit obvious make isn't my expertise. At this point I am a bit confused and cannot seem to find reference on the matter. Help would be appreciated. 

Thank you in advance!

--- a/configure.ac
+++ b/configure.ac
@@ -2695,15 +2695,15 @@ AC_ARG_WITH([libgcrypt],
     if test -f "$withval" && test -x "$withval"; then
       with_libgcrypt_config="$withval"
       with_libgcrypt="yes"
-    else if test -f "$withval/bin/gcrypt-config" && test -x "$withval/bin/gcrypt-config"; then
-      with_libgcrypt_config="$withval/bin/gcrypt-config"
+    else if test -f "$withval/bin/libgcrypt-config" && test -x "$withval/bin/libgcrypt-config"; then
+      with_libgcrypt_config="$withval/bin/libgcrypt-config"
       with_libgcrypt="yes"
     else if test -d "$withval"; then
       GCRYPT_CPPFLAGS="$GCRYPT_CPPFLAGS -I$withval/include"
       GCRYPT_LDFLAGS="$GCRYPT_LDFLAGS -L$withval/lib"
       with_libgcrypt="yes"
     else
-      with_libgcrypt_config="gcrypt-config"
+      with_libgcrypt_config="libgcrypt-config"
       with_libgcrypt="$withval"
     fi; fi; fi
   ],

